File: cypress-framework/cypress/pages/LoginPage.js
class LoginPage {
  static visit() {
    cy.visit("/index.php?route=account/login&language=en-gb");
    return this;
  }
  static fillEmail(v)    { cy.get("#input-email").clear().type(v); return this; }
  static fillPassword(v) { cy.get("#input-password").clear().type(v, { log: false }); return this; }
  static submit()        { cy.get("button[type=submit]").click(); return this; }
  static errorText()     { return cy.get(".alert-danger, #alert .alert-danger"); }
}
export default LoginPage;

File: selenium-framework/pages/login_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage


class LoginPage(BasePage):
    """Storefront /account/login.

    Selectors confirmed against OpenCart 4.1.x
    upload/catalog/view/template/account/login.twig — `#input-email` and
    `#input-password`.
    """
    EMAIL    = (By.ID, "input-email")
    PASSWORD = (By.ID, "input-password")
    SUBMIT   = (By.CSS_SELECTOR, "button[type='submit']")
    ALERT    = (By.CSS_SELECTOR, ".alert-danger, #alert .alert-danger, #account-login .alert-danger")
    FORGOT   = (By.PARTIAL_LINK_TEXT, "Forgotten Password")

    def go(self):
        return self.open("account/login&language=en-gb")

    def login(self, email: str, password: str):
        self.type(self.EMAIL, email)
        self.type(self.PASSWORD, password)
        self.click(self.SUBMIT)
        return self

    def error_text(self) -> str:
        return self.find(self.ALERT).text
